src/Compiler/Microsoft.AspNetCore.Razor.Language/test/IntegrationTests/ComponentCodeGenerationTestBase.cs

```diff
@@ -1709,6 +1709,48 @@ public class MyComponent : ComponentBase
         CompileToAssembly(generated);
     }
 
+    [IntegrationTestFact]
+    public void BindToComponent_SpecifiesValue_WithMatchingProperties_WithNameof()
+    {
+        // Arrange
+        AdditionalSyntaxTrees.Add(Parse(@"
+using System;
+using Microsoft.AspNetCore.Components;
+
+namespace Test
+{
+    public class MyComponent : ComponentBase
+    {
+        [Parameter]
+        public int Value { get; set; }
+
+        [Parameter]
+        public Action<int> ValueChanged { get; set; }
+    }
+}"));
+
+        // Act
+        var generated = CompileToCSharp(@"
+<MyComponent @bind-Value=""ParentValue"" />
```

Comment:
Trying to think about all the quirky behavior with `nameof` and how it could intersect with this feature. 

Ensure `nameof` doesn't bind to a keyword.

```csharp
public class @int : ComponentBase
{
    [Parameter]
    public bool BoolParameter { get; set; }
}
```

```razor
<int BoolParameter="true" />
```

`nameof(dynamic)` cannot bind to the type `dynamic` (late bound type) but can bind to other types named `dynamic`.

```csharp
public class dynamic : ComponentBase
{
    [Parameter]
    public bool BoolParameter { get; set; }
}
```

```razor
<dynamic BoolParameter="true" />
```

Note: all of these samples compile today.

Reply:
Both @int and dynamic work as expected. Added tests to verify.